src/atomWithMutation.ts

```diff
@@ -0,0 +1,56 @@
+import { DocumentInput } from '@urql/core'
+import type { AnyVariables, Client, OperationContext } from '@urql/core'
+import { atom } from 'jotai/vanilla'
+import type { Getter, WritableAtom } from 'jotai/vanilla'
+import { pipe, subscribe } from 'wonka'
+import { clientAtom } from './clientAtom'
+import {
+  type InitialOperationResult,
+  urqlReactCompatibleInitialState,
+} from './common'
+
+export type AtomWithMutation<
+  Data,
+  Variables extends AnyVariables
+> = WritableAtom<
+  InitialOperationResult<Data, Variables>,
+  [Variables, Partial<OperationContext>] | [Variables],
+  Promise<InitialOperationResult<Data, Variables>>
+>
+
+export function atomWithMutation<
+  Data = unknown,
+  Variables extends AnyVariables = AnyVariables
+>(
+  query: DocumentInput<Data, Variables>,
+  getClient: (get: Getter) => Client = (get) => get(clientAtom)
```

Comment:
Hm, but these are arguments...

Reply:
Because all it takes is Document. Similar to react-urql bindings. There is no variables, there is no context, there is document and client. That's it. So making it object would be redundant.

Reply:
If this is how react-urql does, it's fine.

Reply:
Yeah it's.
https://github.com/urql-graphql/urql/blob/35657e771633be43b850f0a1f7fdbac94b075dd7/packages/react-urql/src/hooks/useMutation.ts#L146